src/compose/images.ts

```diff
@@ -598,12 +598,15 @@ export async function inspectByName(imageName: string) {
 
 	// Run the queries in sequence, return the first one that matches or
 	// the error from the last query
-	return await [inspectByURI, inspectByReference, inspectByDigest].reduce(
-		(promise, query) => promise.catch(() => query(imageName)),
-		Promise.reject(
-			'Promise sequence in inspectByName is broken. This is a bug.',
-		),
-	);
+	let err;
+	for (const query of [inspectByURI, inspectByReference, inspectByDigest]) {
+		try {
+			return await query(imageName);
+		} catch ($err) {
+			err = $err;
+		}
+	}
+	throw err;
```

Comment:
Won't this throw every time?

Reply:
Nope, because `return await query(imageName);` will return and exit the function before throwing, so this will throw the final error but only if they all error (preserving previous behavior). That said this logic is generally kinda weird in both cases so I wonder if there's an even clearer way to write it, but I haven't thought of one yet

Reply:
you're right, I was missing the return. I think is debatable that the last error is the one that needs to be returned. Maybe the throw error can be replaced by a 
```suggestion
	new StatusError(
	        404,
	        `Failed to find an image matching ${imageName}`,
        )
```
and just discard the other errors

Reply:
Maybe, but my aim isn't to change the observed error (I have no idea what sort of knock on effect that might have)